### ISSUE TITLE
xcaddy build failed ctx.AppIsConfigured undefined

### ISSUE THREAD

**Issue by cowsay** [gaukas/clienthellod#13]
Tried to build xcaddy with modcaddy 

`xcaddy build --with github.com/gaukas/clienthellod/modcaddy`

Got this:

```
go: downloading github.com/googleapis/enterprise-certificate-proxy v0.2.5
go: downloading cloud.google.com/go/compute v1.20.1
go: downloading github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
2023/10/02 20:58:39 [INFO] exec (timeout=0s): /usr/local/go/bin/go build -o /root/4/caddy -ldflags -w -s -trimpath
# github.com/gaukas/clienthellod/modcaddy/listener
/root/.go/pkg/mod/github.com/gaukas/clienthellod@v0.4.0/modcaddy/listener/listener.go:65:10: ctx.AppIsConfigured undefined (type caddy.Context has no field or method AppIsConfigured)
# github.com/gaukas/clienthellod/modcaddy/handler
/root/.go/pkg/mod/github.com/gaukas/clienthellod@v0.4.0/modcaddy/handler/handler.go:53:10: ctx.AppIsConfigured undefined (type caddy.Context has no field or method AppIsConfigured)
2023/10/02 20:58:40 [INFO] Cleaning up temporary folder: /tmp/buildenv_2023-10-02-2058.860729220
2023/10/02 20:58:40 [FATAL] exit status 1
```

What can be wrong?

**Comment by gaukas** [comment]
It could be a Caddy version problem. 

Could you please try again with `github.com/caddyserver/caddy/v2 v2.6.4`? 

I might need some time to look into this issue. It seems that they renamed the function to `AppIfConfigured` in `v2.7.0`:
https://github.com/caddyserver/caddy/issues/4893
https://github.com/caddyserver/caddy/pull/5397

**Comment by gaukas** [comment]
Or, a PR is very much welcome if it worked with just the function name changed. 

> mholt: Great; yeah, I can 99% guarantee that nobody else is using this function.

https://github.com/caddyserver/caddy/pull/5397#pullrequestreview-1316233068

**Comment by cowsay** [comment]
Yes, managed to build this with rename and type check fix (`xcaddy build v2.6.4` also worked)

**Comment by gaukas** [comment]
Thanks for the PR! I left some suggestions and comments for passing the CI and reducing redundancy, if you'd love to. Or I can do a push, either way.
